core/src/main/java/edu/wpi/grip/core/operations/composite/PublishVideoOperation.java

```diff
@@ -115,8 +113,16 @@ public PublishVideoOperation(InputSocket.Factory inputSocketFactory) {
       server.setSource(serverSource);
 
       ourTable = cameraPublisherTable.getSubTable("GRIP-" + totalStepCount);
-      ourTable.putStringArray("streams",
-          new String[]{"mjpeg:http://" + getHostName() + ":" + ourPort + "/?action=stream"});
+      try {
+        InetAddress localHost = InetAddress.getLocalHost();
+        ourTable.putStringArray("streams",
+            new String[]{
+                generateStreamUrl(localHost.getHostName(), ourPort),
+                generateStreamUrl(localHost.getHostAddress(), ourPort)
+            });
+      } catch (UnknownHostException e) {
+        ourTable.putStringArray("streams", new String[0]);
```

Comment:
Should we log something here?

Reply:
Maybe? The documentation isn't very clear on what circumstances will cause this exception to be thrown. It might never be thrown for the local host since it appears to only happen if the host is unreachable.

Reply:
Interesting. @PeterJohnson do you have any thoughts on what might cause this on the local system or if this is a non-issue?

Reply:
It sounds like how the Java getLocalHost function works is it gets the local hostname, then resolves it into an address.  So it's definitely possible to fail: https://stackoverflow.com/questions/1881546/inetaddress-getlocalhost-throws-unknownhostexception